### ISSUE TITLE
能否增加optimzier 设置 weight decay 方法

### ISSUE THREAD

**Issue by xperzy** [PaddlePaddle/Paddle#39033]
目前optimizer中有set_lr方法可以手动设置lr，
能否增加set_weight_decay方法，用于手动设置weight decay coef，需求是在每个iteration的时候可以手动更改weight decay的coef值

**Comment by xperzy** [comment]
请问使用 `optimizer.regularization = paddle.regularizer.L2Decay(wd)` 直接赋值是否可行？

例如：
```python
for batch_idx, batch_data in enumerate(dataloader):
    optimizer.clear_grad()
    my_coeff = get_wd(batch_idx)  # weight decay scheduler, get target wd for each iteration
    optimizer.regularization = paddle.regularizer.L2Decay(my_coeff) # NOTE: is this correct?
    ...
    loss.backward()
    ...
    optimizer.step()
```

**Comment by liutiexing** [comment]
您好，这个应该可以满足您的需求
